.github/workflows/ci-win.yml

```diff
@@ -51,20 +51,27 @@ jobs:
         path: hadoop-winutils
         ref: d018bd9c919dee1448b95519351cc591a6338a00
 
+    - name: Pull Hadoop
+      # A local Hadoop setup is required on Windows, very sad...
+      uses: nick-fields/retry@v2
+      with:
+        max_attempts: 10
+        timeout_minutes: 5
+        retry_wait_seconds: 3
+        command: |
+          cd $HOME
+          Invoke-WebRequest -Uri https://archive.apache.org/dist/hadoop/common/hadoop-3.3.2/hadoop-3.3.2.tar.gz -OutFile hadoop-3.3.2.tar.gz
+
     - name: Setup Hadoop
       # A local Hadoop setup is required on Windows, very sad...
-      timeout-minutes: 30
       run: |
-        cd $HOME
-        Invoke-WebRequest -Uri https://archive.apache.org/dist/hadoop/common/hadoop-3.3.2/hadoop-3.3.2.tar.gz -OutFile hadoop-3.3.2.tar.gz
-
         echo "::group::Extract hadoop-3.3.2.tar.gz"
         mkdir hadoop-3.3.2
         cd hadoop-3.3.2
         7z e ..\hadoop-3.3.2.tar.gz
         echo "::endgroup::"
-
         echo "HADOOP_HOME=$HOME\hadoop-3.3.2" >> $env:GITHUB_ENV
+
     - name: Add Hadoop winutils
       run: |
         cd hadoop-winutils\hadoop-3.2.2\bin
```

Comment:
just wondering: is this intentionally referring to `3.2.2` while we download hadoop `3.3.2` ?

Reply:
that's hadoop-winutils (see above) - no real idea, but that works